workspace_api/views.py

```diff
@@ -113,6 +127,36 @@ async def create_workspace(
     return {"name": workspace_name}
 
 
+def create_secret(credentials: Dict[str, Any]):
```

Comment:
Create secret is to generic for a name imho. We use secrets for various purposes. The name should denote what kind of secret this is. Like "create_bucket_secret" or similar.